mailpoet/lib/Config/Migrator.php

```diff
@@ -1003,4 +1006,48 @@ private function setDefaultValueForLoadingThirdPartyLibrariesForExistingInstalls
 
     return true;
   }
+
+  private function migrateSegmentDisplaySettingsOnManageSubscriptionPage(): bool {
+    // skip the migration if the DB version is higher than 3.91.1 or is not set (a new installation)
+    if (version_compare($this->settings->get('db_version', '3.102.0'), '3.101.1', '>')) {
+      return false;
+    }
+
+    global $wpdb;
+    $segmentsTable = esc_sql("{$this->prefix}segments");
+
+    // Add display_in_manage_subscription_page column in case it doesn't exist
+    $displayInManageSubscriptionPageColumnExists = $wpdb->get_results($wpdb->prepare("
+      SELECT COLUMN_NAME
+      FROM INFORMATION_SCHEMA.COLUMNS
+      WHERE table_name = %s AND column_name = 'display_in_manage_subscription_page';
+     ", $segmentsTable));
+
+    if (empty($displayInManageSubscriptionPageColumnExists)) {
+      $addNewColumnQuery = "
+        ALTER TABLE `$segmentsTable`
+        ADD `display_in_manage_subscription_page` int(1) NULL DEFAULT 0;
+      ";
+      $wpdb->query($addNewColumnQuery);
+    }
```

Comment:
You've added the column to the table definition and it will be created via `dbDelta`. Here is [the code that does that](https://github.com/mailpoet/mailpoet/blob/357dede83d2c4c8224cc4ee24011946fb47f4c9b/mailpoet/lib/Config/Migrator.php#L84-L87).
So this block of code for adding the column is not needed here.

Reply:
Thanks for the note, @costasovo. 🙌🏿

When testing, the column was not created for me. There was probably an issue with the way I ran the migrator.
I have a question: Does this code have a side effect?

Reply:
> When testing, the column was not created for me. There was probably an issue with the way I ran the migrator.

The migrator runs in 2 cases)
1) On every request, the plugin checks if the version is up to date.
2) During plugin activation

So to trigger it locally, you need to deactivate and reactivate the plugin or go to `mailpoet_settings_table` and decrease the value for `db_version`.

> I have a question: Does this code have a side effect?

I'm not sure what you mean. I would prefer removing the code for adding the column so that we don't have the column definition in two places.